src/Core/ClientTrait.php

```diff
@@ -75,14 +75,17 @@ private function getConnectionType(array $config)
     private function configureAuthentication(array $config)
     {
         $config['keyFile'] = $this->getKeyFile($config);
-
         $this->projectId = $this->detectProjectId($config);
 
+        if (!$config['keyFile'] && !isset($config['credentialsFetcher'])) {
```

Comment:
It will be broken for when the ADC automatically fetches the credentials from metadata server. On GCE and GAE.